OSDOCS-11208 Oc & bare metal bugs for 4.17 RNs (#82508)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

[id="ocp-4-17-bare-metal-hardware-bug-fixes_{context}"]
==== Bare Metal Hardware Provisioning

> * Previously, attempting to configure RAID on specific hardware models by using Redfish might result in the following error: "The attribute StorageControllers/Name is missing from the resource." With this update, the validation logic no longer requires the `Name` field, because it is not mandated by the Redfish standard. (link:https://issues.redhat.com/browse/OCPBUGS-38465[*OCPBUGS-38465*])

Reviewer: Maybe make it even more past tense. 
```suggestion
* Previously, attempting to configure RAID on specific hardware models by using Redfish might have resulted in the following error: "The attribute StorageControllers/Name is missing from the resource." With this update, the validation logic no longer requires the `Name` field, because it is not mandated by the Redfish standard. (link:https://issues.redhat.com/browse/OCPBUGS-38465[*OCPBUGS-38465*])
```

> * Previously, the management interface for `idrac-redfish` in the `redfish’ Bare Metal Operator (BMO) module was incorrectly set to iPXE. This caused the error "Could not find the following interface in the 'ironic.hardware.interfaces.management' entrypoint: ipxe." and deployment failed on Dell Remote Access Controller (iDRAC)-based servers. With this release, the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-37261[*OCPBUGS-37261*])

Reviewer: > the management interface for `idrac-redfish`

Should this be _[iDRAC9 Redfish management interface](https://www.dell.com/support/manuals/en-us/idrac9-lifecycle-controller-v4.x-series/idrac9_4.00.00.00_redfishapiguide_pub/overview?guid=guid-e85fd9c0-f4d1-4eff-be5d-550ebb77ff0d&lang=en-us)_ for clarity? 

At least, it feels like you need a noun after `idrac-redfish`

[id="ocp-4-17-openshift-cli-bug-fixes_{context}"]
==== OpenShift CLI (oc)

> * Previously, when using the delete functionality in the technology preview oc-mirror plug-in v2 feature to remove operator catalogs from mirror registries, the process failed with the following error:

Reviewer: I like the wording in the bug better. 

```suggestion
* Previously, when using `oc-mirror delete` command to remove operator catalogs from mirror registries, the process failed with the following error:
```

+
[source,terminal]
----
2024/08/02 12:18:03 [ERROR]: [OperatorImageCollector] pinging container registry localhost:55000: Get "https://localhost:55000/v2/": http: server gave HTTP response to HTTPS client.
----
+

> This occurred because oc-mirror was querying the local cache using HTTPS instead of HTTP. With this update, the HTTP client is now properly configured before the query, resolving the issue.(link:https://issues.redhat.com/browse/OCPBUGS-41503[*OCPBUGS-41503*])

Reviewer: ```suggestion
This occurred because the `oc-mirror` command was querying the local cache using HTTPS instead of HTTP. With this update, the HTTP client is now properly configured before the query, resolving the issue.(link:https://issues.redhat.com/browse/OCPBUGS-41503[*OCPBUGS-41503*])
```

> * Previously, when using the technology preview oc-mirror plug-in v2 feature in mirror-to-disk mode in disconnected environments, oc-mirror was unable to access `api.openshift.com` to download `graph.tar.gz`, resulting in mirroring failures. With this update, oc-mirror now searches the local cache for the graph image in disconnected environments where the `UPDATE_URL_OVERRIDE` environment variable is set. If the graph image is missing, oc-mirror skips it without failing. (link:https://issues.redhat.com/browse/OCPBUGS-38469[*OCPBUGS-38469*])

Reviewer: Similarly:
```suggestion
* Previously, when using the `oc-mirror` command in mirror-to-disk mode in disconnected environments, the `oc-mirror` command was unable to access `api.openshift.com` to download `graph.tar.gz`, resulting in mirroring failures. With this update, `oc-mirror` now searches the local cache for the graph image in disconnected environments where the `UPDATE_URL_OVERRIDE` environment variable is set. If the graph image is missing, `oc-mirror` skips it without failing. (link:https://issues.redhat.com/browse/OCPBUGS-38469[*OCPBUGS-38469*])
```

> * Previously, when users mirrored operator catalogs with the `--rebuild-catalogs` flag, the catalog cache was regenerated locally, which caused failures either due to compatibility issues with the `opm` binary and the platform or due to cache integrity problems on the cluster. With this update, the `--rebuild-catalogs` flag defaults to true, so catalogs are rebuilt without regenerating the internal cache. Additionally, the image command has been modified to generate the cache during pod startup, which may delay pod initialization. (link:https://issues.redhat.com/browse/OCPBUGS-37667[*OCPBUGS-37667*])

Reviewer: ```suggestion
* Previously, when using the `oc-mirror` command with the `--rebuild-catalogs` flag, the catalog cache was regenerated locally, which caused failures either due to compatibility issues with the `opm` binary and the platform or due to cache integrity problems on the cluster. With this update, the `--rebuild-catalogs` flag defaults to true, so catalogs are rebuilt without regenerating the internal cache. Additionally, the image command has been modified to generate the cache during pod startup, which may delay pod initialization. (link:https://issues.redhat.com/browse/OCPBUGS-37667[*OCPBUGS-37667*])
```

> * Previously, in the technology preview oc-mirror plug-in v2 feature, when running behind a proxy with system proxy settings, oc-mirror did not use the system proxy configuration to recover signatures for releases. With this release, the system proxy settings are now applied during the signature recovery process. (link:https://issues.redhat.com/browse/OCPBUGS-37055[*OCPBUGS-37055*])

Reviewer: ```suggestion
* Previously, when running behind a proxy with system proxy settings, the `oc-mirror` command did not use the system proxy configuration to recover signatures for releases. With this release, the system proxy settings are now applied during the signature recovery process. (link:https://issues.redhat.com/browse/OCPBUGS-37055[*OCPBUGS-37055*])
```

* Previously, {product-title} release images were not visible in certain registries, such as {rhq-short}. This prevented users from installing {product-title} due to the missing release images. With this update, release images are always tagged to ensure they appear in registries like {rhq-short}, enabling proper installation. (link:https://issues.redhat.com/browse/OCPBUGS-36410[*OCPBUGS-36410*])

> * Previously, when using the technology preview oc-mirror plug-in v2 feature in Mirror-to-Disk mode, catalog images and contents were stored in `subfolders` under `working-dir`, based on the image digest. During the Disk-to-Mirror process in fully disconnected environments, oc-mirror tried to resolve the catalog image tag through the source registry, which was unavailable, leading to such errors:

Reviewer: ```suggestion
* Previously, when using the `oc mirror` command in mirror-to-disk mode, catalog images and contents were stored in `subfolders` under `working-dir`, based on the image digest. During the disk-to-mirror process in fully disconnected environments, the `oc-mirror` command tried to resolve the catalog image tag through the source registry, which was unavailable, leading to such errors:
```

Reviewer: Also, maybe move this with the other `oc mirror` bugs?

[source.terminal]
+
----
[ERROR] : [OperatorImageCollector] pinging container registry registry.redhat.io: Get "http://registry.redhat.io/v2/": dial tcp 23.217.255.152:80: i/o timeout
----
+

> With this update, `oc-mirror` checks the local cache during the Disk-to-Mirror process to determine the digest, avoiding the need to query the registry. (link:https://issues.redhat.com/browse/OCPBUGS-36214[*OCPBUGS-36214*])

Reviewer: ```suggestion
With this update, the `oc-mirror` command checks the local cache during the disk-to-mirror process to determine the digest, avoiding the need to query the registry. (link:https://issues.redhat.com/browse/OCPBUGS-36214[*OCPBUGS-36214*])
```

> * Previously, the technology preview oc-mirror plug-in v2 feature failed to mirror operator catalogs from disk-to-mirror in fully disconnected environments. This issue also affected catalogs that specified a `targetCatalog` in the `ImageSetConfiguration` file. With this update, oc-mirror v2 can now successfully mirror catalogs in fully disconnected environments, and the `targetCatalog` functionality works as expected. (link:https://issues.redhat.com/browse/OCPBUGS-34521[*OCPBUGS-34521*])

Reviewer: ```suggestion
* Previously, the `oc-mirror` command failed to mirror operator catalogs from disk-to-mirror in fully disconnected environments. This issue also affected catalogs that specified a `targetCatalog` in the `ImageSetConfiguration` file. With this update, the `oc-mirror` v2 can now successfully mirror catalogs in fully disconnected environments, and the `targetCatalog` functionality works as expected. (link:https://issues.redhat.com/browse/OCPBUGS-34521[*OCPBUGS-34521*])

Also, maybe move this with the other `oc mirror` bugs?
```

> * Previously, the `oc adm must-gather` command took a long time to gather CPU related performance data in large clusters. With this release, the data is gathered in parallel instead of sequentially, which shortens the data collection time. (link:https://issues.redhat.com/browse/OCPBUGS-34360[*OCPBUGS-34360*])

Reviewer: ```suggestion
* Previously, the `oc adm must-gather` command took a long time to gather CPU-related performance data in large clusters. With this release, the data is gathered in parallel instead of sequentially, which shortens the data collection time. (link:https://issues.redhat.com/browse/OCPBUGS-34360[*OCPBUGS-34360*])
```

> * Previously, in the technology preview oc-mirror plug-in v2 feature, errors encountered on registries, such as timeouts, expired authentication tokens, HTTP 500 errors, and so on were not retried automatically. With this update, retries for these errors have been implemented. With this update, retries for these errors are implemented, and users can configure retry behavior with the following flags:

Reviewer: ```suggestion
* Previously, the `oc-mirror` did not automatically perform retries when it encountered errors on registries, such as timeouts, expired authentication tokens, HTTP 500 errors, and so on. With this update, retries for these errors are implemented, and users can configure retry behavior with the following flags:
```

Reviewer: Also, maybe move this with the other `oc mirror` bugs?

+
* `--retry-times`: Specifies the number of retry attempts. Default is 2.
* `--retry-delay`: Sets the delay between retries. Default is 1 second.
* `--image-timeout`: Defines the timeout period for mirroring an image. Default is 10 minutes.

> * `--max-parallel-downloads`: Controls the maximum number of layers to pull simultaneously during a single copy operation. Default is 6.

Reviewer: Not sure this is the best place for this info; should be in the docs, right? Maybe a link here if it is?

Author: It should be, but I'm not seeing it in the flag reference table for oc-mirror plug-in v2. https://docs.openshift.com/container-platform/4.16/installing/disconnected_install/about-installing-oc-mirror-v2.html#oc-mirror-command-reference-v2_about-installing-oc-mirror-v2

Author: I created a jira for them to be added.

Reviewer 2: Can you link me the jira? I can take a look at the docs after GA

+
(link:https://issues.redhat.com/browse/OCPBUGS-34021[*OCPBUGS-34021*])

> * Previously, the technology preview oc-mirror plug-in v2 feature did not generate `ImageDigestMirrorSet` (IDM) or `ImageTagMirrorSet` (ITM) files when mirroring failed due to network issues or invalid operator catalogs. With this update, oc-mirror continues mirroring other images when operator or additional images fail, and only stops when release images fail. Cluster resources are generated based on successfully mirrored images, and all errors are collected in a log file for review.

Reviewer: ```suggestion
* Previously, the `oc-mirror` command did not generate `ImageDigestMirrorSet` (IDMS) or `ImageTagMirrorSet` (ITMS) files when mirroring failed due to network issues or invalid operator catalogs. With this update, the `oc-mirror` continues mirroring other images when operator or additional images fail, and stops only when release images fail. Cluster resources are generated based on successfully mirrored images, and all errors are collected in a log file for review.
```

Also, maybe move this with the other `oc mirror` bugs?

(link:https://issues.redhat.com/browse/OCPBUGS-34020[*OCPBUGS-34020*])

> * Previously, in the technology preview oc-mirror plug-in v2 feature, there was no validation for the `-v2` vs `--v2` flags. As a result, users who mistakenly used `-v2`, that sets the log level to 2 instead of `--v2`, that switches to oc-mirror v2, encountered unclear error messages. With this update, flag validation is provided. If `-v2` is used while the `ImageSetConfig` is using the `v2alpha1` API and `--v2` is not specified, an error message is displayed. The following message is now enabled that provides a clear guidance to the user:

Reviewer: ```suggestion
* Previously, there was no validation for the `-v2` vs `--v2` flags for the `oc-mirror` command. As a result, users who mistakenly used `-v2`, that sets the log level to 2 instead of `--v2`, which switches to `oc-mirror` v2 plugin, causing unclear error messages. With this update, flag validation is provided. If `-v2` is used while the `ImageSetConfig` is using the `v2alpha1` API and `--v2` is not specified, an error message is displayed. The following message is now enabled that provides a clear guidance to the user:
```

+
[source,terminal]
----
[ERROR]: Detected a v2 `ImageSetConfiguration`, please use `--v2` instead of `-v2`.
----
+
(link:https://issues.redhat.com/browse/OCPBUGS-33121[*OCPBUGS-33121*])

> * Previously, the technology preview oc-mirror plug-in v2 feature would stop the mirroring process when it encountered operators using bundle versions that were not compliant with semantic versioning, which also prevented the creation of cluster resources like IDMS, ITMS, and `CatalogSource`. With this fix, oc-mirror v2 now skips these problematic images instead of halting the process. If an image uses incorrect semantic versioning, a warning message is displayed in the console with the relevant image details. (link:https://issues.redhat.com/browse/OCPBUGS-33081[*OCPBUGS-33081*])

Reviewer: ```suggestion
* Previously, oc-mirror plugin v2 would stop the mirroring process when it encountered operators using bundle versions that were not compliant with semantic versioning, which also prevented the creation of cluster resources like IDMS, ITMS, and `CatalogSource` objects. With this fix, the plugin now skips these problematic images instead of halting the process. If an image uses incorrect semantic versioning, a warning message is displayed in the console with the relevant image details. (link:https://issues.redhat.com/browse/OCPBUGS-33081[*OCPBUGS-33081*])
```

> * Previously, the `oc set env` command wrongly changed the apiVersion of Route and `DeploymentConfig` objects, removing the group part. For example, `apps.openshift.io/v1` became `v1`. This caused the command to exit with "unable to recognize no matches for kind" errors. With this release, the error is fixed so that the `os set env` command keeps the correct version in the Route and `DeploymentConfig` objects. (link:https://issues.redhat.com/browse/OCPBUGS-32108[*OCPBUGS-32108*])

Reviewer: ```suggestion
* Previously, the `oc set env` command incorrectly changed the API version of `Route` and `DeploymentConfig` objects, for example, `apps.openshift.io/v1` became `v1`. This caused the command to exit with _unable to recognize no matches for kind_ errors. With this release, the error is fixed so that the `os set env` command keeps the correct API version in `Route` and `DeploymentConfig` objects. (link:https://issues.redhat.com/browse/OCPBUGS-32108[*OCPBUGS-32108*])
```

> * Previously, when a `must-gather` failed for any reason and the user manually deleted the leftover namespace, a cluster role binding created by the must-gather command would remain in the cluster. With this release, when the temporary must-gather namespace is deleted, the associated cluster role binding is automatically deleted with it. (link:https://issues.redhat.com/browse/OCPBUGS-31848[*OCPBUGS-31848*])

Reviewer: ```suggestion
* Previously, when a `must-gather` failed for any reason and the user manually deleted the leftover namespace, a cluster role binding created by the `must-gather` command would remain in the cluster. With this release, when the temporary `must-gather` namespace is deleted, the associated cluster role binding is automatically deleted with it. (link:https://issues.redhat.com/browse/OCPBUGS-31848[*OCPBUGS-31848*])
```

* Previously, when using oc-mirror plugin v2 in mirror-to-disk mode in disconnected environments, the plugin was unable to access `api.openshift.com` to download `graph.tar.gz`, resulting in mirroring failures. With this update, the plugin now searches the local cache for the graph image in disconnected environments where the `UPDATE_URL_OVERRIDE` environment variable is set. If the graph image is missing, the plugin skips it without failing. (link:https://issues.redhat.com/browse/OCPBUGS-38469[*OCPBUGS-38469*])

* Previously, oc-mirror plugin v2 failed to mirror operator catalogs from disk-to-mirror in fully disconnected environments. This issue also affected catalogs that specified a `targetCatalog` in the `ImageSetConfiguration` file. With this update, the plugin can now successfully mirror catalogs in fully disconnected environments, and the `targetCatalog` functionality works as expected. (link:https://issues.redhat.com/browse/OCPBUGS-34521[*OCPBUGS-34521*])

> * Previously, with the oc-mirror plugin v2, there was no validation for the `-v2` vs `--v2` flags for the `oc-mirror` command. As a result, users who mistakenly used `-v2`, that sets the log level to 2 instead of `--v2`, which switches to oc-mirror plugin v2, causing unclear error messages. With this update, flag validation is provided. If `-v2` is used while the `ImageSetConfig` is using the `v2alpha1` API and `--v2` is not specified, an error message is displayed. The following message is now enabled that provides a clear guidance to the user:

Reviewer 2: ```suggestion
* Previously, with the oc-mirror plugin v2, there was no validation for the `-v2` vs `--v2` flags for the `oc-mirror` command. As a result, users who mistakenly used `-v2`, which sets the log level to 2, instead of `--v2`, which switches to oc-mirror plugin v2, received unclear error messages. With this update, flag validation is provided. If the `-v2` flag is used while the `ImageSetConfig` is using the `v2alpha1` API and `--v2` is not specified, an error message is displayed. The following message is now enabled that provides a clear guidance to the user:
```

+
[source,terminal]
----
[ERROR]: Detected a v2 `ImageSetConfiguration`, please use `--v2` instead of `-v2`.
----
+
(link:https://issues.redhat.com/browse/OCPBUGS-33121[*OCPBUGS-33121*])

* Previously, oc-mirror plugin v2 did not automatically perform retries when it encountered errors on registries, such as timeouts, expired authentication tokens, HTTP 500 errors, and so on. With this update, retries for these errors are implemented, and users can configure retry behavior with the following flags:
+

> * `--retry-times`: Specifies the number of retry attempts. Default is 2.
> * `--retry-delay`: Sets the delay between retries. Default is 1 second.
> * `--image-timeout`: Defines the timeout period for mirroring an image. Default is 10 minutes.
> * `--max-parallel-downloads`: Controls the maximum number of layers to pull simultaneously during a single copy operation. Default is 6.

Reviewer 2: these list items should be level 2 so that they're nested under their respective RN:
```suggestion
** `--retry-times`: Specifies the number of retry attempts. Default is 2.
** `--retry-delay`: Sets the delay between retries. Default is 1 second.
** `--image-timeout`: Defines the timeout period for mirroring an image. Default is 10 minutes.
** `--max-parallel-downloads`: Controls the maximum number of layers to pull simultaneously during a single copy operation. Default is 6.
```

+
(link:https://issues.redhat.com/browse/OCPBUGS-34021[*OCPBUGS-34021*])

* Previously, when using the oc-mirror plugin v2 with the `--rebuild-catalogs` flag, the catalog cache was regenerated locally, which caused failures either due to compatibility issues with the `opm` binary and the platform or due to cache integrity problems on the cluster. With this update, the `--rebuild-catalogs` flag defaults to true, so catalogs are rebuilt without regenerating the internal cache. Additionally, the image command has been modified to generate the cache during pod startup, which may delay pod initialization. (link:https://issues.redhat.com/browse/OCPBUGS-37667[*OCPBUGS-37667*])

* Previously, the oc-mirror plugin v2 did not use the system proxy configuration to recover signatures for releases when running behind a proxy with system proxy settings. With this release, the system proxy settings are now applied during the signature recovery process. (link:https://issues.redhat.com/browse/OCPBUGS-37055[*OCPBUGS-37055*])

> * Previously, oc-mirror plugin v2 would stop the mirroring process when it encountered operators using bundle versions that were not compliant with semantic versioning, which also prevented the creation of cluster resources like IDMS, ITMS, and `CatalogSource` objects. With this fix, the plugin now skips these problematic images instead of halting the process. If an image uses incorrect semantic versioning, a warning message is displayed in the console with the relevant image details. (link:https://issues.redhat.com/browse/OCPBUGS-33081[*OCPBUGS-33081*])

Reviewer 2: ```suggestion
* Previously, oc-mirror plugin v2 would stop the mirroring process when it encountered Operators using bundle versions that were not compliant with semantic versioning, which also prevented the creation of cluster resources like IDMS, ITMS, and `CatalogSource` objects. With this fix, the plugin now skips these problematic images instead of halting the process. If an image uses incorrect semantic versioning, a warning message is displayed in the console with the relevant image details. (link:https://issues.redhat.com/browse/OCPBUGS-33081[*OCPBUGS-33081*])
```

> * Previously, oc-mirror plugin v2 did not generate `ImageDigestMirrorSet` (IDMS) or `ImageTagMirrorSet` (ITMS) files when mirroring failed due to network issues or invalid operator catalogs. With this update, the `oc-mirror` continues mirroring other images when operator or additional images fail, and stops only when release images fail. Cluster resources are generated based on successfully mirrored images, and all errors are collected in a log file for review. (link:https://issues.redhat.com/browse/OCPBUGS-34020[*OCPBUGS-34020*])

Reviewer 2: ```suggestion
* Previously, oc-mirror plugin v2 did not generate `ImageDigestMirrorSet` (IDMS) or `ImageTagMirrorSet` (ITMS) files when mirroring failed due to network issues or invalid Operator catalogs. With this update, the `oc-mirror` continues mirroring other images when Operator or additional images fail, and stops only when release images fail. Cluster resources are generated based on successfully mirrored images, and all errors are collected in a log file for review. (link:https://issues.redhat.com/browse/OCPBUGS-34020[*OCPBUGS-34020*])
```

> * Previously, {product-title} release images were not visible in certain registries, such as {rhq-short}. This prevented users from installing {product-title} due to the missing release images. With this update, release images are always tagged to ensure they appear in registries like {rhq-short}, enabling proper installation. (link:https://issues.redhat.com/browse/OCPBUGS-36410[*OCPBUGS-36410*])

Reviewer 2: rhq-short isn't resolving to anything. Based on the history of the Jira issue, it looks like this was supposed to be Quay:
```suggestion
* Previously, {product-title} release images were not visible in certain registries, such as {quay}. This prevented users from installing {product-title} due to the missing release images. With this update, release images are always tagged to ensure they appear in registries like {quay}, enabling proper installation. (link:https://issues.redhat.com/browse/OCPBUGS-36410[*OCPBUGS-36410*])
```

* Previously, the `oc adm must-gather` command took a long time to gather CPU-related performance data in large clusters. With this release, the data is gathered in parallel instead of sequentially, which shortens the data collection time. (link:https://issues.redhat.com/browse/OCPBUGS-34360[*OCPBUGS-34360*])

> * Previously, the `oc set env` command incorrectly changed the API version of `Route` and `DeploymentConfig` objects, for example, `apps.openshift.io/v1` became `v1`. This caused the command to exit with _unable to recognize no matches for kind_ errors. With this release, the error is fixed so that the `os set env` command keeps the correct API version in `Route` and `DeploymentConfig` objects. (link:https://issues.redhat.com/browse/OCPBUGS-32108[*OCPBUGS-32108*])

Reviewer 2: I feel like this should probably be monospace rather than italic:
```suggestion
* Previously, the `oc set env` command incorrectly changed the API version of `Route` and `DeploymentConfig` objects, for example, `apps.openshift.io/v1` became `v1`. This caused the command to exit with `unable to recognize no matches for kind` errors. With this release, the error is fixed so that the `os set env` command keeps the correct API version in `Route` and `DeploymentConfig` objects. (link:https://issues.redhat.com/browse/OCPBUGS-32108[*OCPBUGS-32108*])
```

> * Previously, when a `must-gather` failed for any reason and the user manually deleted the leftover namespace, a cluster role binding created by the `must-gather` command would remain in the cluster. With this release, when the temporary `must-gather` namespace is deleted, the associated cluster role binding is automatically deleted with it. (link:https://issues.redhat.com/browse/OCPBUGS-31848[*OCPBUGS-31848*])

Reviewer 2: ```suggestion
* Previously, when a `must-gather` operation failed for any reason and the user manually deleted the leftover namespace, a cluster role binding created by the `must-gather` command would remain in the cluster. With this release, when the temporary `must-gather` namespace is deleted, the associated cluster role binding is automatically deleted with it. (link:https://issues.redhat.com/browse/OCPBUGS-31848[*OCPBUGS-31848*])
```

> * Previously, when using the `--v2` flag, if no images were mirrored and some were skipped, empty `imds.yaml` and `itms.yaml` files were generated. With this release, the custom resource generation is only triggered when at least one image is successfully mirrored, preventing the creation of empty files. (link:https://issues.redhat.com/browse/OCPBUGS-33775[*OCPBUGS-33775*])

Reviewer 2: ```suggestion
* Previously, when using the `--v2` flag with the oc-mirror plugin v2, if no images were mirrored and some were skipped, empty `imds.yaml` and `itms.yaml` files were generated. With this release, the custom resource generation is only triggered when at least one image is successfully mirrored, preventing the creation of empty files. (link:https://issues.redhat.com/browse/OCPBUGS-33775[*OCPBUGS-33775*])
```
